### PR TITLE
[SPARK-39512] Document docker image release steps

### PR DESCRIPTION
**Pull request by holdenk**
Document the docker image release steps for the release manager to follow when finalizing the release.

### CODE REVIEW

File: site/sitemap.xml [hunk 1]
```diff
@@ -941,27 +941,27 @@
   <changefreq>weekly</changefreq>
 </url>
 <url>
-  <loc>https://spark.apache.org/graphx/</loc>
+  <loc>https://spark.apache.org/news/</loc>
```
Comment (srowen) [inline site/sitemap.xml]: I don't know which ordering is correct, but maybe revert this change?

Comment (srowen) [reply site/sitemap.xml]: @holdenk just checking if you saw this - do we want to revert or is this the right order now?

Comment (gengliangwang) [reply site/sitemap.xml]: +1 @srowen. The changes on this file seem not necessary.

**Comment by gengliangwang** [comment]
@holdenk I followed the steps and it works!
I have built docker images on https://hub.docker.com/u/gengliangwang
If @MaxGekk doesn't have permission to publish it, I can do it for him this time.

**Comment by gengliangwang** [comment]
BTW, I think we should add the docker images into the RC vote email and let the community test them as well.

**Comment by holdenk** [comment]
Maybe, unlike maven repos though we don't have a staging location set up, I think we could ask ASF Infra to make us a staging location?

**Comment by gengliangwang** [comment]
> Maybe, unlike maven repos though we don't have a staging location set up, I think we could ask ASF Infra to make us a staging location?

We can publish RC images with a different tag, e.g. v3.4.0-rc1.
After release, the images can be deleted.

**Comment by gengliangwang** [comment]
FYI I just published docker images for Spark 3.3. release
https://hub.docker.com/r/apache/spark
https://hub.docker.com/r/apache/spark-py
https://hub.docker.com/r/apache/spark-r

I will do send an email to the dev/user list if no issues found during the weekend
cc @holdenk

**Comment by holdenk** [comment]
Sounds good. Want me to add the instructions to the RC step too then?

**Comment by gengliangwang** [comment]
> Sounds good. Want me to add the instructions to the RC step too then?

Sure 👍

File: release-process.md [hunk 2]
```diff
@@ -35,6 +35,9 @@ If you are a new Release Manager, you can read up on the process from the follow
 - gpg for signing https://www.apache.org/dev/openpgp.html
 - svn https://www.apache.org/dev/version-control.html#https-svn
 
+
+You should also get access to the ASF Dockerhub, this is not required for the release candidates but can take some time. You can get access by filing a INFRA JIRA ticket.
```
Comment (tgravescs) [inline release-process.md]: do we want to add a link to infra jira, or instructions, or example ticket someone already did this with to make this a bit easier on someone new?

File: release-process.md [hunk 3]
```diff
@@ -387,6 +390,16 @@ $ git shortlog v1.1.1 --grep "$EXPR" > contrib.txt
...
 
+<h4>Create and upload Spark Docker Images</h4>
+
+The Spark docker images are created using the `./bin/docker-image-tool.sh` that is included in the release artifacts.
+
+
+You should install `docker buildx` so that you can cross-compile for multiple archs as ARM is becoming increasing popular. If you have access to both an ARM and an x86 machine you should set up a [remote builder as described here](https://scalingpythonml.com/2020/12/11/some-sharp-corners-with-docker-buildx.html), but if you only have one [docker buildx with QEMU works fine as we don't use cgo](https://docs.docker.com/buildx/working-with-buildx/).
+
+
+Once you have your cross-platform docker build environment setup, extract the build artifact (e.g. `tar -xvf spark-3.3.0-bin-hadoop3.tgz`), go into the director (e.g. `cd spark-3.3.0-bin-hadoop3`) and build the containers and publish them to the Spark dockerhub (e.g. `./bin/docker-image-tool.sh -r docker.io/apache -p ./kubernetes/dockerfiles/spark/bindings/python/Dockerfile -t v3.3.0 -X -b java_image_tag=11-jre-slim build`)
```
Comment (tgravescs) [inline release-process.md]: spelling of directory

File: site/release-process.html [hunk 4]
```diff
@@ -497,6 +499,14 @@ <h4>Update the Spark website</h4>
...
+
+<p>You should install <code class="language-plaintext highlighter-rouge">docker buildx</code> so that you can cross-compile for multiple archs as ARM is becoming increasing popular. If you have access to both an ARM and an x86 machine you should set up a <a href="https://scalingpythonml.com/2020/12/11/some-sharp-corners-with-docker-buildx.html">remote builder as described here</a>, but if you only have one <a href="https://docs.docker.com/buildx/working-with-buildx/">docker buildx with QEMU works fine as we don&#8217;t use cgo</a>.</p>
+
+<p>Once you have your cross-platform docker build environment setup, extract the build artifact (e.g. <code class="language-plaintext highlighter-rouge">tar -xvf spark-3.3.0-bin-hadoop3.tgz</code>), go into the director (e.g. <code class="language-plaintext highlighter-rouge">cd spark-3.3.0-bin-hadoop3</code>) and build the containers and publish them to the Spark dockerhub (e.g. <code class="language-plaintext highlighter-rouge">./bin/docker-image-tool.sh -r docker.io/apache -p ./kubernetes/dockerfiles/spark/bindings/python/Dockerfile -t v3.3.0 -X -b java_image_tag=11-jre-slim build</code>)</p>
```
Comment (tgravescs) [inline site/release-process.html]: spelling directory

**Comment by srowen** [comment]
Ping @holdenk

**Comment by holdenk** [comment]
My bad let me pick this up tomorrow.

**Comment by holdenk** [comment]
ping @MaxGekk & @tgravescs @gengliangwang since y'all had comments on the first draft, this one looking ok?

**Review by tgravescs — APPROVED** [review]
thanks Holden

**Review by MaxGekk — APPROVED** [review]

**Review by gengliangwang — APPROVED** [review]
Thanks for the instructions!

**Review by dongjoon-hyun — APPROVED** [review]

**Review by xinrong-meng — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to asf-site

**Comment by dongjoon-hyun** [comment]
Thank you, @holdenk and all. According to this guideline, Apache Spark 3.2.2 images are also published too.

Screen Shot 2022-07-19 at 9 27 31 AM
